File: backend/app/schemas/violation.py
"""Violation response schemas."""

from datetime import datetime
from typing import Any

from pydantic import BaseModel


class ViolationOut(BaseModel):
    id: int
    client_id: str
    transaction_id: str | None
    violation_type: str
    severity: str
    message: str
    details: dict[str, Any] | None
    detected_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/violation.ts
export interface Violation {
  id: number;
  client_id: string;
  transaction_id: string | null;
  violation_type: "short_selling" | "concentration_risk" | "day_trading";
  severity: "error" | "warning";
  message: string;
  details: Record<string, unknown> | null;
  detected_at: string;
}
